eo-runtime/src/main/java/EOorg/EOeolang/EOtxt/EOsprintf.java

```diff
@@ -65,13 +70,63 @@ public EOsprintf(final Phi sigma) {
                     final String format = new Param(rho, "format").strong(String.class);
                     final Phi[] args = new Param(rho, "args").strong(Phi[].class);
                     final Collection<Object> items = new LinkedList<>();
-                    for (final Phi arg : args) {
-                        items.add(new Dataized(arg).take());
+                    for (int idx = 0; idx < args.length; idx += 1) {
+                        items.add(format(args[idx], idx, format));
                     }
                     return new Data.ToPhi(String.format(Locale.US, format, items.toArray()));
                 }
             )
         );
     }
 
+    /**
+     * Return formatted object, if argumen is byte array.
+     *
+     * @param phi Argument of the sprintf
+     * @param index Index of the argument in array
+     * @param format Format string
+     * @return Formatted argument
+     * @todo #1049:30min Current implementation supports
+     *  only character formatting. It means that byte array
+     *  become an first bytes character string. We need to
+     *  support another flags, like %a, %o, %s, etc.
+     */
+    private static Object format(final Phi phi, final int index, final String format) {
+        final Object arg = new Dataized(phi).take();
+        Object result = arg;
+        if (arg instanceof byte[]) {
+            final int occurrence = findOccurrence(format, index);
+            final String flag = format.substring(occurrence, occurrence + 2);
+            if (CHAR_FLAG.equalsIgnoreCase(flag)) {
+                result =  (char) ((byte[]) arg)[0];
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Find index of the flag occurrence.
+     *
+     * @param format Format string
+     * @param index Index of the argument in array
+     * @return Index of th flag
+     */
+    private static int findOccurrence(final String format, final int index) {
+        final String flag = "%";
+        String fmt = format;
+        int result = -1;
+        for (int occurrence = 0; occurrence <= index; occurrence += 1) {
```

Comment:
@Graur I think you can use `++occurrence` here

Reply:
@EugeneDar if I do so, I get checkstyle's error [IllegalTokenCheck](https://checkstyle.sourceforge.io/apidocs/com/puppycrawl/tools/checkstyle/checks/coding/IllegalTokenCheck.html):  Using '++' is not allowed.